packages/i18nify-js/src/modules/phoneNumber/getMaskedPhoneNumber.ts

```diff
@@ -0,0 +1,136 @@
+import getDialCodeByCountryCode from './getDialCodeByCountryCode';
+import { withErrorBoundary } from '../../common/errorBoundary';
+import PHONE_FORMATTER_MAPPER from './data/phoneFormatterMapper.json';
+import {
+  cleanPhoneNumber,
+  detectCountryAndDialCodeFromPhone,
+  replaceFirstXsWithChars,
+  replaceLastXsWithChars,
+} from './utils';
+import { GetMaskedPhoneNumberOptions } from './types';
+
+/**
+ * Generates a masked phone number based on provided options.
+ * This function handles the complexity of different phone number formats and
+ * masking preferences such as complete masking or partial masking of digits.
+ *
+ * @param {GetMaskedPhoneNumberOptions} options - Options for generating the masked phone number.
+ * @param {CountryCodeType} options.countryCode - The country code associated with the phone number.
+ * @param {boolean} options.withDialCode - Determines if the dial code should be included in the masked number.
+ * @param {string} options.phoneNumber - The actual phone number to mask.
+ * @param {MaskingOptions} options.maskingOptions - Options to specify how the masking should be performed.
+ * @returns {string} The masked phone number formatted as per the specified options.
+ * @throws {Error} Throws an error if both countryCode and phoneNumber are empty or if other input validations fail.
+ */
+const getMaskedPhoneNumber = ({
+  countryCode,
+  withDialCode = true,
+  phoneNumber,
+  maskingOptions = {
+    maskingStyle: 'full',
+    maskedDigitsCount: 0,
+    maskingChar: 'x',
+  },
+}: GetMaskedPhoneNumberOptions) => {
+  if (!countryCode && !phoneNumber) {
+    throw new Error('Either countryCode and phoneNumber is mandatory.');
+  }
+
+  let maskedContactNumber: string;
+
+  if (phoneNumber) {
+    // Clean the phone number to remove any non-numeric characters, except the leading '+'
+    let updatedPhoneNumber = phoneNumber;
+    updatedPhoneNumber = updatedPhoneNumber.toString();
+    updatedPhoneNumber = cleanPhoneNumber(updatedPhoneNumber);
+
+    // Detect the country code and dial code from the cleaned phone number
+    const countryData = detectCountryAndDialCodeFromPhone(updatedPhoneNumber);
+    const updatedCountryCode = countryCode || countryData.countryCode;
+
+    // Get the phone number formatting template based on the country code
+    let formattingTemplate = PHONE_FORMATTER_MAPPER[updatedCountryCode];
+
+    // In case phone number doesn't have dialCode masking should happen without formatting
+    if (!formattingTemplate) {
+      return updatedPhoneNumber.replace(
+        /./g,
+        maskingOptions.maskingChar || 'x',
```

Comment:
maskingChar should have a default value X at the top.. we need not do OR operation everytime.